nemo_skills/inference/reward_model.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+import json
+import logging
+import sys
+from copy import deepcopy
+from dataclasses import asdict, field
+from pathlib import Path
+
+import hydra
+from tqdm import tqdm
+
+from nemo_skills.code_execution.sandbox import get_sandbox, sandbox_params
+from nemo_skills.inference.server.code_execution_model import ErrorRecoveryConfig, server_params
+from nemo_skills.inference.server.reward_model import RequestException, get_reward_model
+from nemo_skills.prompt.utils import get_prompt
+from nemo_skills.utils import get_fields_docstring, get_help_message, nested_dataclass, setup_logging
+
+LOG = logging.getLogger(__file__)
+
+
+@nested_dataclass(kw_only=True)
+class RewardModelConfig:
+    """LLM reward model parameters."""
+
+    output_file: str | None = None  # Where to save the generations
+    # Inference server configuration {server_params} {error_recovery_params}
+    server: dict = field(default_factory=dict)
+    # Sandbox configuration {sandbox_params}
+    sandbox: dict = field(default_factory=dict)
+    # Prompt configuration - path to yaml files
+    prompt_template: str | None = None  # not required for OpenAI server
+    prompt_config: str | None = None  # we will fetch it from dataset dir if not provided
+    examples_type: str | None = None  # to be able to customize few-shot examples
+
+    # Can specify one of the existing datasets.
+    dataset: str | None = None
+    split: str | None = None  # Can be train, validation, test or train_full (train + validation)
+    input_file: str | None = None  # Can directly specify an input file, if using a custom dataset
+    input_dir: str | None = None  # Can specify an input direct
+    output_dir: str | None = None
+    random_seed: str | None = None
+
+    batch_size: int = 128
+    max_samples: int = -1  # If > 0, will stop after generating this many samples. Useful for debugging
+    skip_filled: bool = False  # If True, will skip the generations that are already in the output file
+
+    # if > 0, will skip this many samples from the beginning of the data file.
+    # Useful if need to run multiple slurm jobs on the same data file
+    offset: int = 0
+
+    generation_key: str = "generation"
```

Comment:
Is this currently unused? If so, would be good to make sure this is used to control what is the key we save the scores in with default set to whatever you currently save it to

Reply:
Fixed in ed3f676b14a7e6342c1658df7dda95ad7ab7b1e5